components/navigation/DesktopNav.tsx

```diff
@@ -1,16 +1,105 @@
-import React from 'react';
-import { usePathname } from 'next/navigation';
+'use client';
 
-//Components
+// Components
 import NavigationLinkContainer from './NavigationLinkContainer';
 import AboutRouteIcon from '../icons/navigation/AboutRouteIcon';
 import AddItemRouteIcon from '../icons/navigation/AddItemRouteIcon';
 import SearchRouteIcon from '../icons/navigation/SearchRouteIcon';
 import ProfileRouteIcon from '../icons/navigation/ProfileRouteIcon';
 import MessageRouteIcon from '../icons/navigation/MessageRouteIcon';
+import NotificationDot from '../NotificationDot';
+
+import { usePathname } from 'next/navigation';
+import { useEffect, useState } from 'react';
+import selectUserUnreadConversations from '@/supabase/models/messaging/selectUserUnreadConversations';
+import newClient from '@/supabase/utils/newClient';
+import selectLoggedUserId from '@/supabase/utils/selectLoggedUserId';
 
 const DesktopNav = () => {
   const pathname = usePathname();
+  const supabase = newClient();
+  const [userId, setUserId] = useState<string>('');
+  const [hasNotification, setHasNotification] = useState<boolean>(false);
+
+  useEffect(() => {
+    const getUserId = async () => {
+      try {
+        const data = await selectLoggedUserId();
+        if (data) {
+          setUserId(data);
+        } else {
+          console.error('Failed to fetch user Id');
+        }
+      } catch (error) {
+        console.error('Error fetching user Id:', error);
```

Comment:
This seems redundant  - your `selectLoggedUserId` function already throws an error so you'll get it in the catch - just throw it again